docs/chapters/architecture/SOTA-system.rst

```diff
@@ -0,0 +1,614 @@
+| PELUX Developer       | Maintains the            | Wants to configure the    |
+|                       | distribution             | SOTA Client for new use   |
+|                       |                          | cases                     |
++-----------------------+--------------------------+---------------------------+
+| BSP Developer         | Adapts PELUX to new      | Wants to integrate the    |
+|                       | platforms                | SOTA system on new        |
+|                       |                          | platforms                 |
++-----------------------+--------------------------+---------------------------+
+| UI Developer          | Develops a user          | Wants to integrate SOTA   |
+|                       | interface on PELUX       | Client to the GUI         |
++-----------------------+--------------------------+---------------------------+
+
+Constraints
+-----------
+
+The SOTA system should:
+* work on a typical embedded Linux context, for instance with a bootloader such
+as U-Boot or a build system such as Yocto
+* be free software
+
+System scope and context
+------------------------
+
+The purpose of this section is to put the SOTA system into a broader context,
+and show how this subsystem interacts with other parts of the larger PELUX
+system.
+
+Business context
+^^^^^^^^^^^^^^^^
+
+The business context shows how the SOTA system uses other subsystems and is
+being used by other software.
+
+.. blockdiag::
+
+    blockdiag {
+        node_width = 160;
+
+        class emphasis [color = pink];
+
+        orientation=portrait;
+
+        upd8db [label = "Update database", shape = flowchart.database];
+        sotasyst [label = "SOTA System", class = "emphasis"];
+        gui [label = "GUI"];
+        ecusfpgas [label = "ECUs/FPGAs"];
+        partitions [label = "Partitions"];
+        bootloader [label = "Bootloader"];
+
+        upd8db <-> sotasyst [label = "Requests", folded];
+
+        gui -> sotasyst [label = "Triggers"];
+
+        sotasyst -> ecusfpgas [label = "Updates"];
+        sotasyst -> partitions [label = "Updates"];
+        sotasyst -> bootloader [label = "Configures"];
+    }
+
++------------------+-----------------------------------------------------------+
+| Neighbor         | Description                                               |
++==================+===========================================================+
+| Updates database | Used by the SOTA Server to store update artifacts and     |
+|                  | meta-data                                                 |
++------------------+-----------------------------------------------------------+
+| GUI              | Uses the SOTA Client to trigger an update                 |
++------------------+-----------------------------------------------------------+
+| Bootloader       | Used by the SOTA Client to atomically update a system     |
```

Comment:
Isn't the bootloader mainly used to select the system to boot (in a partition A/B setup, for instance)? Does it also perform the actual update?